js_modules/dagster-ui/packages/ui-core/src/assets/catalog/AssetCatalogV2VirtualizedTable.tsx

```diff
@@ -1,240 +1,219 @@
-                  {content}
-                </div>
-              );
+  const totalHeight = rowVirtualizer.getTotalSize();
+  const items = rowVirtualizer.getVirtualItems();
+
+  return (
+    <Container ref={containerRef}>
+      <Inner $totalHeight={totalHeight}>
+        <div
+          style={{
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            width: '100%',
+            transform: `translateY(${items[0]?.start ?? 0}px)`,
+          }}
+        >
+          {items.map(({index}) => {
+            const item = rowItems[index];
+
+            if (!item) {
+              return null;
+            }
 
-              if ('shimmer' in item) {
-                return wrapper(
+            if ('shimmer' in item) {
+              return (
+                <div key={index} data-index={index} ref={rowVirtualizer.measureElement}>
                   <Box padding={{top: 12, horizontal: 20}}>
                     <Skeleton $height={30} $width="100%" />
-                  </Box>,
-                );
-              }
+                  </Box>
+                </div>
+              );
+            }
+
+            if ('header' in item) {
+              const assetsInGroup =
+                grouped[item.group]?.assets.map((asset) => tokenForAssetKey(asset.key)) ?? [];
 
-              if ('header' in item) {
-                const assetsInGroupJSON = groupedByStatus[item.status].map((asset) =>
-                  JSON.stringify(asset.key.path),
-                );
+              const checkedState = assetsInGroup.every((asset) => checkedDisplayKeys.has(asset))
+                ? 'checked'
+                : assetsInGroup.some((asset) => checkedDisplayKeys.has(asset))
+                  ? 'indeterminate'
+                  : 'unchecked';
 
-                const checkedState = assetsInGroupJSON.every((asset) =>
-                  checkedDisplayKeys.has(asset),
-                )
-                  ? 'checked'
-                  : assetsInGroupJSON.some((asset) => checkedDisplayKeys.has(asset))
-                    ? 'indeterminate'
-                    : 'unchecked';
+              const Klass = grouped[item.group]?.renderGroupHeader;
 
-                return (
-                  <div key={key} data-index={index} ref={rowVirtualizer.measureElement}>
-                    <StatusHeader
-                      status={item.status}
-                      open={openStatuses.has(item.status)}
-                      assets={groupedByStatus[item.status]}
-                      onToggleChecked={onToggleGroup(item.status)}
+              return (
+                <div key={item.group} data-index={index} ref={rowVirtualizer.measureElement}>
+                  {Klass ? (
+                    <Klass
+                      group={item.group}
+                      open={!closedGroups.has(item.group)}
+                      assets={grouped[item.group].assets}
+                      onToggleChecked={onToggleGroup(item.group)}
```

Comment:
Hmm got tripped up by this for a hot second because I didn't realize `onToggleGroup` is a factory method. I wonder if we could call this `buildOnToggleGroup`, I don't think we really have a convention for it.